### PR TITLE
Add objective-c compatibility / example

### PR DESCRIPTION
**Pull request by bsneed**
- Added wrappers to support ObjC basics, such as track/identify/flush/etc.
- Added objc example app.
- Added objc example app to CI build process.
- Added tests to verify non-trivial pass-through methods.

### CODE REVIEW

File: Examples/apps/ObjCExample/ObjCExample/AppDelegate.m [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+@end
+
+@implementation AppDelegate
+
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    // Override point for customization after application launch.
+    SEGConfiguration *config = [[SEGConfiguration alloc] initWithWriteKey:@"WRITE_KEY"];
+    config.trackApplicationLifecycleEvents = TRUE;
```
Comment (migs647) [inline Examples/apps/ObjCExample/ObjCExample/AppDelegate.m:22]: TRUE == YES

File: Sources/Segment/ObjC/ObjCAnalytics.swift [hunk 2]
```diff
@@ -0,0 +1,161 @@
+//
+//  File.swift
```
Comment (migs647) [inline Sources/Segment/ObjC/ObjCAnalytics.swift:2]: File.swift

File: Sources/Segment/ObjC/ObjCConfiguration.swift [hunk 3]
```diff
@@ -0,0 +1,132 @@
+//
+//  File.swift
```
Comment (migs647) [inline Sources/Segment/ObjC/ObjCConfiguration.swift:2]: File.swift

**Review by migs647 — APPROVED** [review]
LGTM with bool fix

**Review by alanjcharles — APPROVED** [review]
thanks for walking me through! Looks good

**Pull request merged**